### PR TITLE
Feature/list of opportunities back end

### PR DESCRIPTION
**Pull request by mattwills8**
Sorry this took some time...

I've merged master so this branch is up to date

There ended up being a few separate changes here and I have a habit of babbling so I'll try to describe them as concisely as I can:

**1. Seeding the database.**
I've added functionality for seeding the database so we can all work with the same data. In the database dir you'll see the data for the users and opportunities to be seeded. 
Just visit the route /seed/users to seed users, and seed/opps to seed the opportunities

**2. Updated user schema.**
I've added an array property for storing the list of id's of opportunities applied for. Ideally here we'd store the mongoose id obejct of the opportunity, but a string seems to work okay too.

**3. Some new endpoints for front end retrieval and to easily view the data (couldnt find a decent mongo gui, you can type these into the browser to see the data, or use Postman)**
GET /api/all-users - sget all users in the db
GET /api/user/id - gets user with given id
GET /api/all-opps - get all opportunities
GET /api/opp/id - shows opp with given id
POST /api/opps - gets opportunities from an array of ids passed in the body (used for profile page)

**4. Changes to login redux state**
I've changed the log in and sign up actions so that the whole user object gets passed around in the redux state rather than just the email. This means we only have to make one new request (for the user's opps) rather than two when displaying the profile page. Only issue here is we need to find a way to encode the password (or just not send it through the login action I guess).

**5. Profile page displaying user's applied for opportunities.**
The way to get this to show (currently we dont have functionality to let a user apply for an opp within the app), is to:
reset the db
first seed the db with opps (/seed/opps)
copy some opp ids into the applied for array in the user data to be seeded (seedUsers.js I think)
then seed the users (/seed/users/)
now login and visit the profile page of the user you added the opp ids to, should see them

### CODE REVIEW

**Comment by mattwills8** [comment]
changes have now been incorporated from @Mongolialady 's branch. At this point a user can apply for opportunities and have them appear on their profile page, with everything updated in the databse

File: client/src/actions/index.js [hunk 1]
```diff
@@ -19,14 +34,70 @@ function signup ({ email, password }) {
   return async dispatch => {
     const response = await signupApiClient.signup({ email, password })
     if (response.status === 200) {
-      return dispatch(login({ email }))
+      return response.json()
+        .then(user => {
+          dispatch(login(user))
+        })
```
Comment (bbmoz) [inline client/src/actions/index.js]: Opt for async/await here, like so:
```js
const user = await response.json()
dispatch(login(user))
```

File: client/src/actions/index.js [hunk 2]
```diff
@@ -19,14 +34,70 @@ function signup ({ email, password }) {
...
+function getOppsAppliedFor (user) {
+  return async dispatch => {
+    const response = await oppsApiClient.getOppsAppliedFor(user.opportunitiesAppliedFor)
+    if (response.status === 200) {
+      return response.json().then(opps => {
+        dispatch({
+          type: GET_OPPS_APPLIED_FOR,
+          payload: opps
+        })
+      })
```
Comment (bbmoz) [inline client/src/actions/index.js]: See previous comment about using async/await! :)

File: server/controllers/loginController.test.js [hunk 3]
```diff
@@ -15,17 +15,15 @@ test('login with email and password', t => {
...
       }
     }
   }
 
   // mocks
   const resMock = mock(res)
-    .expects('sendStatus')
-    .once()
-    .withExactArgs(200)
+    .expects('send')
```
Comment (bbmoz) [inline server/controllers/loginController.test.js]: Add the following to the top and remove the `const res = ...` in all tests for both the `loginController.test.js` and `signupController.test.js` files.

```js
let res

test.beforeEach(() => {
  res = {
    sendStatus () {},
    setHeader () {},
    send () {}
  }
})
```

Also, modify the `signupController.test.js` when the `res` is mocked for the `signup new user` test to be:
```js
const mockRes = mock(res)
  .expects('send')
  .once()
  .withExactArgs(JSON.stringify({ savedUser: { email: 'user@email.com' } }))
```

**Comment by bbmoz** [comment]
@timdavish @mattwills8 Tests seem to be failing due to the change in the way we are sending data in our signup and login controllers from this PR. I added a comment that should fix the test errors! See above. 👍 I'll merge once these are in.

**Comment by mattwills8** [comment]
Cheers guys! I’ll get those changes in ASAP :)

**Pull request merged**